### DISCUSSION TITLE
Recorder bug

### DISCUSSION THREAD

**Discussion by mazezone** [zepp-health/zeppos-forum#301] [Q&A] [answered]
Hi!
Lines:
```
import { create, id, codec } from '@zos/media'
const recorder = create(id.RECORDER)

```
Causing the simulator and real clocks (GTR4 /ZeppOS 3) to completely freeze. Is it a bug?
On the watch, this happens after exiting the mini app.

**Comment by silver-zepp** [comment]
Hey @mazezone . The media library is currently unsupported by the Simulator so you will have to use a dummy object like this:
```js
// just a test (doesn't work on the Simulator yet)
import { SoundPlayer } from "../include/libs/easy-media"
// actual sound object (to use with a physical device)
//const player = new SoundPlayer("click.mp3", true);
// dummy object (to use on a simulator)
const player = { play: function() {}, destroy: function() {} }; 
```
To address the freezing further, try using this library instead and see if the behavior persists: https://github.com/silver-zepp/zeppos-easy-media
Make sure to execute `player.destroy()` in your `onDestroy()` method.

**Reply by StarRrScreaM** [reply]
Hello @silver-zepp. Tried the standard method and your library to play the sound on a GTR4 clock with zepp 3.0, accordingly there are a couple of issues:
- When declaring a player and giving it a playback file, it plays only 1 time!!! I can't play it again.
`const player = new SoundPlayer("media/beep.mp3", true);`
`player.play();`
Then I try to reproduce it again, because the path is already specified and the file is cached? And nothing's happening...
`player.play();`
- referring to your example from the library, I prescribe a change of file and play it again - this approach works, but leads to the clock hangs after 1 - 2 clicks on the button.
`player.changeFile('media/beep.mp3');`
`player.play();`

How to play the same file several times (e.g. "beep-beep"), and avoid hangs when changing the audio source.

**Reply by mazezone** [reply]
Hi @silver-zepp !
Thanks for the answer. I saw your library for Media.
The problem is that the smartwatch freezes when you exit the mini app. This only applies to the Recorder. 
Player is working fine.
Is it a bug?

**Reply by silver-zepp** [reply, marked as answer]
Oh, I see @mazezone. This very well might be a bug. Just make sure that you are stopping the recorder and deregistering all the callbacks in `onDestroy()` or prior. If that's the case, then it's for sure a bug and we'll have to look into it as soon as possible.

@StarRrScreaM thanks for the report as well. Try removing the true flag from the `SoundPlayer` constructor. This should get rid of the lag you experience but you'll have to make sure that the files that you are trying to play are as short as possible, otherwise when another sound overlaps the ongoing one, it will be ignored. And if you still experience bug with sound being played only once then we might have some new bug around. Try the [AutoGUI's](https://github.com/silver-zepp/zeppos-autogui) calculator example, just make sure to use the actual SoundPlayer class instead of a dummy one and see if each click retriggers the sound play (just don't rush with button clicks, in case you ain't using the `stop_on_change` flag).
![image](https://github.com/zepp-health/zeppos-forum/assets/138894963/5df58395-30d4-4002-b62b-e59cb411be7c)

**Reply by mazezone** [reply]
Thanks @silver-zepp ! In the case of Record, there are no events that need to be canceled. That's all there is to it: recorder declaration, file/codec name setting and two commands: start and stop. The two lines I wrote are enough for a bug.

Thank you! I'm waiting for the fix.

**Reply by StarRrScreaM** [reply]
> ...Try the [AutoGUI's](https://github.com/silver-zepp/zeppos-autogui) calculator example...

doesn't work ... only once ... I tried with "true" and "false", still nothing.

https://github.com/zepp-health/zeppos-forum/assets/158058564/37e6ba63-76ce-46fd-8984-57e60dcebd2f
